Link generator handles non-git sources

sean-morris opened the pull request:
The link generator creates gitpuller links for non-git
sources including archives from google drive, dropbox
and any publicly accessible web url.

sean-morris committed:
Link generator handles non-git sources

The link generator creates gitpuller links for non-git
sources including archives from google drive, dropbox
and any publicly accessible web URL. In order to handle a non-
git source correctly in nbgitpuller, we added the provider
parameter to the URL string.

Other Changes:
- if git is the source, the branch parameter is always included on the
URL and defaults to 'main' if none is provided. This should then allow
users that have not upgraded their nbgitpuller version to still
work.
- All sources(git and non-git) can handle a path to a particular folder
or notebook. I had taken this ability away from non-git sources in
earlier versions.

sean-morris committed:
fixed conflicts

sean-morris committed:
Changed value for Web choice to generic_web

We renamed the nbgitpuller-standard-web downloader to
nbgitpuller-generic-web. This commit reflects this change in the
link generator.

sean-morris committed:
Changed query parameter provider to content_provider

This syncs the changes made in nbgitpuller app

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -1,19 +1,25 @@
+        url.searchParams.set('branch', "main");
     }
-
+    
```
```suggestion

```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -22,20 +28,26 @@ function generateRegularUrl(hubUrl, urlPath, repoUrl, branch) {
 
     nextUrlParams.append('repo', repoUrl);
-
+    
```
```suggestion

```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -141,28 +163,60 @@ function changeTab(div) {
 function displayLink() {
     var form = document.getElementById('linkgenerator');
-
+    
```
```suggestion

```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -141,28 +163,60 @@ function changeTab(div) {
         var activeTab = document.querySelector(".nav-link.active").id;
-
+        var source = form.querySelector('input[name="source"]:checked').value;
+        
```
```suggestion

```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -269,7 +373,9 @@ function main() {
       }
     }
 
+    
```
```suggestion
```

consideRatio commented on link.html:
```diff
@@ -221,16 +221,16 @@ <h1>nbgitpuller link generator<a class="headerlink" href="#nbgitpuller-link-gene
-
-        <div class="form-group row">
-          <label for="repo" class="col-sm-2 col-form-label">Git Repository URL</label>
+        
```
```suggestion

```

consideRatio commented on link.html:
```diff
@@ -252,26 +252,100 @@ <h1>nbgitpuller link generator<a class="headerlink" href="#nbgitpuller-link-gene
         </div>
-
-        <div class="form-group row" id="content-repo-group" hidden="true">
+        
```
```suggestion

```

consideRatio commented on link.html:
```diff
@@ -252,26 +252,100 @@ <h1>nbgitpuller link generator<a class="headerlink" href="#nbgitpuller-link-gene
+            </div>
+          </div>
+        </div>
+        
+        
+        
```
```suggestion

```

consideRatio commented on link.html:
```diff
@@ -252,26 +252,100 @@ <h1>nbgitpuller link generator<a class="headerlink" href="#nbgitpuller-link-gene
-              <input name="content-branch" id="content-branch" type="text" class="form-control" value="master" aria-label="Branch Name" aria-describedby="content-branch-prepend-label">
+              <input name="content-branch" id="content-branch" type="text" placeholder="main" aria-label="Branch Name" aria-describedby="content-branch-prepend-label">
+              <small id="default-content-message" class="form-text text-muted">
+                 If left blank, the default branch of your repository is used. Also note, Github now names the default branch <code>main</code> instead of <code>master</code> on
+                 <a href="https://github.blog/changelog/2020-10-01-the-default-branch-for-newly-created-repositories-is-now-main/">
+                 new GitHub repositories</a>
```
```suggestion
                If left blank, the default branch of your repository is used. Also note, Github now names the default branch <code>main</code> instead of <code>master</code> on
                <a href="https://github.blog/changelog/2020-10-01-the-default-branch-for-newly-created-repositories-is-now-main/">
                new GitHub repositories</a>
```

consideRatio commented on link.html:
```diff
@@ -252,26 +252,100 @@ <h1>nbgitpuller link generator<a class="headerlink" href="#nbgitpuller-link-gene
+            </div>
+          </div>
+        </div>
+        
```
```suggestion

```

consideRatio commented on link.html:
```diff
@@ -252,26 +252,100 @@ <h1>nbgitpuller link generator<a class="headerlink" href="#nbgitpuller-link-gene
+            </div>
+          </div>
+        </div>
+        
```
```suggestion

```

consideRatio commented on link.html:
```diff
@@ -252,26 +252,100 @@ <h1>nbgitpuller link generator<a class="headerlink" href="#nbgitpuller-link-gene
+            </div>
+          </div>
+        </div>
+        
```
```suggestion

```

consideRatio commented on link.html:
```diff
@@ -282,7 +356,7 @@ <h1>nbgitpuller link generator<a class="headerlink" href="#nbgitpuller-link-gene
           </div>
         </div>
-
+        
```
```suggestion

```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -246,11 +340,21 @@ function render() {
+            );
+        }
+    )
+    
```
```suggestion

```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -225,6 +317,7 @@ function render() {
     var form = document.getElementById('linkgenerator');
     var appName = form.querySelector('input[name="app"]:checked').value;
 
+    
```
```suggestion
```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -213,6 +278,33 @@ function populateFromQueryString() {
     }
 }
 
+function hideShowByClassName(cls, hideShow){
+  [].forEach.call(document.querySelectorAll(cls), function (el) {
+    el.style.display = hideShow;
+    setDisabled = (hideShow == 'none')
+    $(el).find("input").each(function(){
+        $(this).prop("disabled", setDisabled);
+    });
+  });
+}
```
```suggestion
function hideShowByClassName(cls, hideShow){
    [].forEach.call(document.querySelectorAll(cls), function (el) {
        el.style.display = hideShow;
        setDisabled = (hideShow == 'none')
        $(el).find("input").each(function(){
            $(this).prop("disabled", setDisabled);
        });
    });
}
```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -171,26 +225,37 @@ function displayLink() {
         }
-
+        args = {
+          "source": source,
+          "contentRepoUrl": contentRepoUrl,
+          "contentRepoBranch": contentRepoBranch,
+          "driveUrl": driveUrl,
+          "dropUrl": dropUrl,
+          "webUrl": webUrl
```
```suggestion
            "source": source,
            "contentRepoUrl": contentRepoUrl,
            "contentRepoBranch": contentRepoBranch,
            "driveUrl": driveUrl,
            "dropUrl": dropUrl,
            "webUrl": webUrl
```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -141,28 +163,60 @@ function changeTab(div) {
-            var repoName = generateCloneDirectoryName(repoUrl);
+            var repoName = generateCloneDirectoryName(contentRepoUrl);
+            if(source !== "git"){
+              repoName = ""
```
```suggestion
                repoName = ""
```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -141,28 +163,60 @@ function changeTab(div) {
 }
 
+function handleSource(args){
+  source = args["source"];
+  branch = "";
+  compressed = true;
+  sourceUrl ="";
+  if(source == "git"){
+      sourceUrl = args["contentRepoUrl"];
+      branch = args["contentRepoBranch"];
+      compressed = false;
+  } else if(source == "googledrive"){
+      sourceUrl = args["driveUrl"];
+  } else if(source == "dropbox"){
+      sourceUrl =  args["dropUrl"];
+  } else if(source == "generic_web"){
+      sourceUrl =  args["webUrl"];
+  }
+  return {
+    "branch": branch,
+    "sourceUrl": sourceUrl,
+    "compressed": compressed
+  }
```
```suggestion
    source = args["source"];
    branch = "";
    compressed = true;
    sourceUrl ="";
    if(source == "git"){
        sourceUrl = args["contentRepoUrl"];
        branch = args["contentRepoBranch"];
        compressed = false;
    } else if(source == "googledrive"){
        sourceUrl = args["driveUrl"];
    } else if(source == "dropbox"){
        sourceUrl =  args["dropUrl"];
    } else if(source == "generic_web"){
        sourceUrl =  args["webUrl"];
    }
    return {
        "branch": branch,
        "sourceUrl": sourceUrl,
        "compressed": compressed
    }
```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -141,28 +163,60 @@ function changeTab(div) {
  */
 function generateCloneDirectoryName(gitCloneUrl) {
+    if(gitCloneUrl.slice(-1) == "/")
+      gitCloneUrl = gitCloneUrl.slice(0,-1);
```
```suggestion
        gitCloneUrl = gitCloneUrl.slice(0,-1);
```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -100,37 +118,41 @@ var apps = {
-        env_repo_branch.pattern = ".+";
-        content_repo.hidden = false;
-        content_branch.hidden = false;
+        
```
```suggestion
```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -100,37 +118,41 @@ var apps = {
+        
+        env_repo_group.style.display = '';
+        env_repo.disabled = false;
+
```
```suggestion
```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -100,37 +118,41 @@ var apps = {
     var id = div.id;
-
+    var form = document.getElementById('linkgenerator');
+    
```
```suggestion

```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -50,20 +62,26 @@ function generateCanvasUrl(hubUrl, urlPath, repoUrl, branch) {
+    if(compressed) {
+      nextUrlParams.append('content_provider', source);
+    }
+    
```
```suggestion

```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -50,20 +62,26 @@ function generateCanvasUrl(hubUrl, urlPath, repoUrl, branch) {
     nextUrlParams.append('repo', contentRepoUrl);
 
+    if(compressed) {
+      nextUrlParams.append('content_provider', source);
```
```suggestion
        nextUrlParams.append('content_provider', source);
```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -22,20 +28,26 @@ function generateRegularUrl(hubUrl, urlPath, repoUrl, branch) {
+    if(compressed) {
+      nextUrlParams.append('content_provider', source);
+    }
+    
```
```suggestion

```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -22,20 +28,26 @@ function generateRegularUrl(hubUrl, urlPath, repoUrl, branch) {
-
+    
+    if(compressed) {
+      nextUrlParams.append('content_provider', source);
```
```suggestion
        nextUrlParams.append('content_provider', source);
```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -1,19 +1,25 @@
+    if(compressed) {
+      url.searchParams.set('content_provider', source);
+    }
+    
```
```suggestion

```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -1,19 +1,25 @@
 
     url.searchParams.set('repo', repoUrl);
-
+    
```
```suggestion

```

consideRatio commented on .circleci/config.yml:
```diff
@@ -0,0 +1,12 @@
+version: 2.1
+jobs:
+  build:
+    docker:
+      - image: circleci/python:3.6
+    steps:
+      - run: echo "no operation build"
+
+workflows:
+  build:
+    jobs:
+      - build
```
This file seems like a dummy file of some kind commited by mistake. If you need to add CI logic, go for github action based logic.

sean-morris replied:
I had added it because circleci was throwing an error related to not finding a config file when I submitted the PR. I think I copied it from the main branch.  So it should just be removed altogether?

consideRatio replied:
Yeah remove it, ill absolutely ignore such error and not let that block merge.

consideRatio committed:
Fix indentation and remove all spaces on otherwise empty lines

Whenever an empty line is replaced with an empty line with spaces, it will make the `git diff` views in terminals and GitHub UI cluttered.

The indentation in .js files were inconsistent and this commit also fixes that.

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -246,11 +328,20 @@ function render() {
+            element.addEventListener('change', function(){
+                displaySource();
+                render();
+              }
+            );
```
```suggestion
            });
```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -238,6 +319,7 @@ function render() {
             document.getElementById('filepath').disabled = false;
         }
     }
+
```
```suggestion
```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -100,37 +112,38 @@ var apps = {
+    document.getElementById('canvas-link').value = "";
+}
+
+
```
```suggestion
```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -1,19 +1,22 @@
+        url.searchParams.set('branch', "main");
     }
-
     if (!url.pathname.endsWith('/')) {
```
```suggestion

    if (!url.pathname.endsWith('/')) {
```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -1,19 +1,22 @@
     url.searchParams.set('repo', repoUrl);
-
+    if(compressed) {
+      url.searchParams.set('content_provider', source);
```
```suggestion
        url.searchParams.set('content_provider', source);
```

consideRatio commented on link.html:
```diff
@@ -221,16 +221,15 @@ <h1>nbgitpuller link generator<a class="headerlink" href="#nbgitpuller-link-gene
               Must be a valid git URL
             </div>
-            <small class="form-text text-muted" id="env-repo-help-text" hidden="true">
+            <small class="form-text text-muted" id="env-repo-help-text" >
```
```suggestion
            <small class="form-text text-muted" id="env-repo-help-text">
```

consideRatio commented on link.html:
```diff
@@ -252,26 +251,93 @@ <h1>nbgitpuller link generator<a class="headerlink" href="#nbgitpuller-link-gene
           </div>
           <div class="col-sm-4">
-            <div class="input-group" id="content-branch-group" hidden="true">
+            <div class="input-group" id="content-branch-group" >
```
```suggestion
            <div class="input-group" id="content-branch-group">
```

consideRatio commented on link.html:
```diff
@@ -252,26 +251,93 @@ <h1>nbgitpuller link generator<a class="headerlink" href="#nbgitpuller-link-gene
+              <input name="content-branch" id="content-branch" type="text" placeholder="main" aria-label="Branch Name" aria-describedby="content-branch-prepend-label">
+              <small id="default-content-message" class="form-text text-muted">
+                If left blank, the default branch of your repository is used. Also note, Github now names the default branch <code>main</code> instead of <code>master</code> on
+                <a href="https://github.blog/changelog/2020-10-01-the-default-branch-for-newly-created-repositories-is-now-main/">
+                new GitHub repositories</a>
```
```suggestion
                <a href="https://github.blog/changelog/2020-10-01-the-default-branch-for-newly-created-repositories-is-now-main/">new GitHub repositories</a>
```

consideRatio commented on link.html:
```diff
@@ -252,26 +251,93 @@ <h1>nbgitpuller link generator<a class="headerlink" href="#nbgitpuller-link-gene
+          <label for="drive-url" class="col-sm-2 col-form-label">Google Drive Shared Link</label>
+          <div class="col-sm-10">
+            <input class="form-control" type="text" id="drive-url" placeholder="https://drive.google.com/file/d/1p3m0h5UGWdLkVVP0S1HpG2yeDlU/view?usp=sharing"
+              oninput="displayLink()" required pattern="https?:\/\/drive.google.com.+">
```
In this case, it won't matter I'm almost 100% confident, but if this pattern is used when it really must match the drive.google.com urls but nothing else then one should beware. This pattern could match an url like `drive.google.com-my-malicious-domain.com` as well.

sean-morris replied:
ahhh. ok. yes. I will add /file/.+ to the end of the pattern

consideRatio commented on link.html:
```diff
@@ -282,7 +348,6 @@ <h1>nbgitpuller link generator<a class="headerlink" href="#nbgitpuller-link-gene
           </div>
         </div>
-
         <div class="form-group row" id="app-container">
```
```suggestion

        <div class="form-group row" id="app-container">
```

consideRatio committed:
Fix additional formatting details

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -100,37 +113,37 @@ var apps = {
     }
 }
 
+function clearLinks(){
+    document.getElementById('default-link').value = "";
+    document.getElementById('binder-link').value = "";
+    document.getElementById('canvas-link').value = "";
+}
+
 function changeTab(div) {
     var hub = document.getElementById("hub");
     var hub_help_text = document.getElementById("hub-help-text");
-    var env_repo = document.getElementById("repo");
-    var env_repo_branch = document.getElementById("branch");
-    var env_repo_help_text = document.getElementById("env-repo-help-text");
-    var content_repo = document.getElementById("content-repo-group");
-    var content_branch = document.getElementById("content-branch-group");
+    var env_repo_group = document.getElementById("env-repo-group");
+    var env_repo = document.getElementById("env-repo");
```
I understand you have renamed the element id's to match the variable names - okay, it was a bit out of scope for this PRs main purpose and makes the review and tracking changes a bit trickier, but sure I appreciate the consistency.

Removed: `content_repo` `content_branch` env_repo_help_text` and `env_repo_branch`, 
Added: `env_repo_group`

Ah... You have refactored so you hide an entire `<div>` of things in one go? Okay makes sense.

sean-morris replied:
ok I hear you here. It got a bit tricky to follow the patterns in these pages which inspired me to make things clearer at least to myself! I can see how that makes the PR review more difficult. I will make sure to separate out refactoring commits from commits directly related to the PR.

sean-morris replied:
The main problem I ran into with the original code was that we were changed the meaning of repoX(e.g. repoName, repoURL, etc)  depending on the context which caused me great headaches!  In other words, if you were NOT on the binder tab then repoX meant the content repo where the notebooks are stored. But if you were on the binder tab, then repoX meant the environment git repo you are using and contentRepoX meant the repo with the notebooks. It killed me when we went to add the non-git sources to the binder tab. 

You will see logic in the next commit that removes the need a conditional we used to generate the urlPath because of this situation(link.js - line 245)

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -141,28 +154,58 @@ function changeTab(div) {
  * See https://github.com/git/git/blob/1c52ecf4ba0f4f7af72775695fee653f50737c71/builtin/clone.c#L276
  */
 function generateCloneDirectoryName(gitCloneUrl) {
+    if(gitCloneUrl.slice(-1) == "/")
+        gitCloneUrl = gitCloneUrl.slice(0,-1);
     var lastPart = gitCloneUrl.split('/').slice(-1)[0];
     return lastPart.split(':').slice(-1)[0].replace(/(\.git|\.bundle)?/, '');
 }
```
I don't understand this change, can you help me by describing what you are doing here?

Preferably, this function would already have a documentation string to clarify its purpose and typical input it can expect and should be able to handle. I assume you are making this function support a new kind of input by this change - is that correct?

sean-morris replied:
I was responding to what happens if you add a forward slash to the end of the git clone URL. The function tries to split by forward slash and you end up without the name you wanted. I watched it happen when we were running training sessions for Community College users. Probably should have done the separate commit for the change or not included it here. Pull it out?

consideRatio replied:
If this can be a 2-line change, and its unrelated to tests etc, then yeah ok pull it out - i dont want you to need to invest much time in it further otherwise though.

I think overall, reading splice/split/index etc is hard to read, so having a comment clarifying what goes on can be helpful.

sean-morris replied:
ok I left it in and added some documentation

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -141,28 +154,58 @@ function changeTab(div) {
     return lastPart.split(':').slice(-1)[0].replace(/(\.git|\.bundle)?/, '');
 }
 
+function handleSource(args){
```
Since this returns structured data from incoming structured data, I think the name `handleSource` isn't very clear. Could you make it more explicit and write a short docstring?

I understand from code inspection that this takes some kind of `args`, which I guess relates to what has been passed via a form submission or similar, and then that it parses those into another structure.

Hmmm, perhaps "parse" would make sense to include in the function name, maybe "form data", maybe also "into ..."? For a helper function that is only used internally etc, having a long descriptive name is something I find very helpful.

Could you write a sentence or two about this as a docstring, and then tweak the name of the function?

sean-morris replied:
I renamed this to : configureContentProviderAttrs and added some documentation

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -141,28 +154,58 @@ function changeTab(div) {
+    branch = "";
+    compressed = true;
+    sourceUrl ="";
+    if(source == "git"){
+        sourceUrl = args["contentRepoUrl"];
+        branch = args["contentRepoBranch"];
+        compressed = false;
+    } else if(source == "googledrive"){
+        sourceUrl = args["driveUrl"];
+    } else if(source == "dropbox"){
+        sourceUrl =  args["dropUrl"];
+    } else if(source == "generic_web"){
+        sourceUrl =  args["webUrl"];
+    }
```
Seeing this, I'm thinking "aha, source is the choice of content provider". Perhaps naming this content provider type or similar would be a more suitable name. Not sure about the full picture here, so maybe source makes sense.

sean-morris replied:
yes the original piece referred to "git" as the source. I changed everything to be content_provider(or some derivative) and got rid of "source" in both the javascript and HTML attributes

sean-morris committed:
Implemented changes from PR review

- Changed "source" to "content-provider" in both files
- In link.html, added detail to regEx for URL patterns
- removed circleci folder I thought was needed
- In generateBinderUrl, clarified the parameter names to reflect
envRepoX and contentRepoX and fixed the generatation of the URL
- In binder tab, the environment repo now includes the branch if not
specified
- added documentation
- cleaned up the generation of the URLPath

sean-morris committed:
fixed conflicts

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -1,20 +1,20 @@
     url.searchParams.set('repo', repoUrl);
     if(compressed) {
-        url.searchParams.set('content_provider', source);
+      url.searchParams.set('content_provider', contentProvider);
```
```suggestion
        url.searchParams.set('content_provider', contentProvider);
```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -26,23 +26,23 @@ function generateRegularUrl(hubUrl, urlPath, repoUrl, branch, compressed, source
     nextUrlParams.append('repo', repoUrl);
     if(compressed) {
-        nextUrlParams.append('content_provider', source);
+      nextUrlParams.append('content_provider', contentProvider);
```
```suggestion
        nextUrlParams.append('content_provider', contentProvider);
```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -57,32 +57,35 @@ function generateCanvasUrl(hubUrl, urlPath, repoUrl, branch, compressed, source)
 
     if(compressed) {
-        nextUrlParams.append('content_provider', source);
+      nextUrlParams.append('content_provider', contentProvider);
```
```suggestion
        nextUrlParams.append('content_provider', contentProvider);
```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -191,53 +213,51 @@ function displayLink() {
-            "driveUrl": driveUrl,
-            "dropUrl": dropUrl,
-            "webUrl": webUrl
+          "contentProvider": contentProvider,
+          "contentGitRepoUrl": contentGitRepoUrl,
+          "contentGitRepoBranch": contentGitRepoBranch,
+          "driveUrl": driveUrl,
+          "dropUrl": dropUrl,
+          "webUrl": webUrl
```
```suggestion
            "contentProvider": contentProvider,
            "contentGitRepoUrl": contentGitRepoUrl,
            "contentGitRepoBranch": contentGitRepoBranch,
            "driveUrl": driveUrl,
            "dropUrl": dropUrl,
            "webUrl": webUrl
```

consideRatio commented on link.html:
```diff
@@ -350,7 +357,7 @@ <h1>nbgitpuller link generator<a class="headerlink" href="#nbgitpuller-link-gene
         <div class="form-group row" id="app-container">
           <div class="col-sm-2 col-form-label">
-            <label for="app" class=>Application to Open</label>
+            <label for="app-container" class=>Application to Open</label>
             <small class="form-text text-muted">
             </small>
```
```suggestion
            <label>Application to Open</label>
```

consideRatio commented on link.html:
```diff
@@ -251,40 +252,44 @@ <h1>nbgitpuller link generator<a class="headerlink" href="#nbgitpuller-link-gene
-            <label for="app" class=>Notebook Source</label>
+            <label for="app-content-provider" class=>Notebook Content Provider</label>
             <small class="form-text text-muted">
             </small>
```
```suggestion
            <label>Notebook Content Provider</label>
```

consideRatio commented on _static/link_gen/link.js:
```diff
@@ -160,27 +169,40 @@ function generateCloneDirectoryName(gitCloneUrl) {
+      contentProviderURL =  args["webUrl"];
+  }
+  return {
+    "branch": branch,
+    "contentProviderURL": contentProviderURL,
+    "compressed": compressed
```
```suggestion
      "branch": branch,
      "contentProviderURL": contentProviderURL,
      "compressed": compressed
```

sean-morris committed:
Apply suggestions from code review

Co-authored-by: Erik Sundell <erik.i.sundell@gmail.com>

sean-morris commented:
My spacing is off. I will check that out. What is the standard here?

consideRatio commented on link.html:
```diff
@@ -253,25 +253,97 @@ <h1>nbgitpuller link generator<a class="headerlink" href="#nbgitpuller-link-gene
-        <div class="form-group row" id="content-repo-group" hidden="true">
+        <div class="form-group row" id="app-content-provider">
+          <div class="col-sm-2 col-form-label">
+            <label for="app-content-provider" class=>Notebook Content Provider</label>
+            <small class="form-text text-muted">
+            </small>
```
```suggestion
            <label>Notebook Content Provider</label>
```

sean-morris replied:
are you wanting to get rid of the for="xxx" here? The "class=" is gone yes.

consideRatio commented:
> My spacing is off. I will check that out. What is the standard here?

I think 2 for html files, 4 for python, and 4 for js files. That seems to be the spacing of choice in this repo, it will always be 4 for python files across the JuptyerHub organization, but I'm not sure there is a well defined standard for js/html files, so then I'm just now making sure the code stays consistent within files and between files in the repo.

sean-morris committed:
Update link.html

Co-authored-by: Erik Sundell <erik.i.sundell@gmail.com>

sean-morris committed:
Cleaned up HTML errors

- removed duplicate data-X on nav bar button
- removed duplicate id on the source repo button and download button

sean-morris committed:
Merge branch 'gh-pages-non-git' of https://github.com/sean-morris/nbgitpuller into gh-pages-non-git

sean-morris committed:
Query param content_provider now contentProvider

consideRatio commented on link.html:
```diff
@@ -130,7 +130,7 @@ <h3 id="searchlabel">Quick search</h3>
 
 <div class="dropdown-buttons-trigger">
-    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
+    <button id="sourc-repo-dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
```
Typo on source.

Not sure about the purpose of this change, because there was a duplicate id? Is the id needed at all on these buttons?

sean-morris replied:
they did not look needed but I didn't think I should just remove it.

sean-morris replied:
fixed the misspelling
